### DISCUSSION TITLE
Does the element of the slice have to be a structure

### DISCUSSION THREAD

**Discussion by lirui001** [antonmedv/expr#482] [Q&A] [answered]
In my code，i want to use any check slice data，but my slice is about []string，i don't kown how to get element about slice in my expression.

In your example：
type Tweet struct {
	Len int
}

type Env struct {
	Tweets []Tweet
}

func main() {
	code := `all(Tweets, {.Len <= 240})`

	program, err := expr.Compile(code, expr.Env(Env{}))
	if err != nil {
		panic(err)
	}

	env := Env{
		Tweets: []Tweet{{42}, {98}, {69}},
	}
	output, err := expr.Run(program, env)
	if err != nil {
		panic(err)
	}

	fmt.Println(output)
}

In my example：
type Env struct {
	Tweets []int
}

func main() {
	code := `all(Tweets, {.<= 240})`

	program, err := expr.Compile(code, expr.Env(Env{}))
	if err != nil {
		panic(err)
	}

	env := Env{
		Tweets: []int{42, 98, 69},
	}
	output, err := expr.Run(program, env)
	if err != nil {
		panic(err)
	}

	fmt.Println(output)
}

the code compile error：Error compiling expression: expected name

**Comment by antonmedv** [comment, marked as answer]
Use #
